archesproject/arches
# Enables two factor authentication

### Types of changes


### Description of Change


This update allows administrators to enable/force two-factor authentication in their Arches application. 

The testable workflows of this PR are:
- to test settings.ENABLE_TWO_FACTOR_AUTHENTICATION
  - create a new user & register via email
  - login as new user
  - navigate to user profile page
  - update preferences, then click link to send 2fa update email
  - via 2fa update email, access 2fa settings page
  - test that the logic is correct when generating a new QR code, manually entering the registration data, and disabling 2fa ( all should force a logout )
  - login to test 2fa
  - navigate back to user profile page
  - update preferences, then click link to send 2fa update email
  - disable 2fa 
  - login to test that 2fa is disabled
- to test settings.FORCE_TWO_FACTOR_AUTHENTICATION
  - create a new user or work with user already in system
  - attempt to login, after first login page this should redirect to a sent email query page
  - via email, access 2fa settings
  - enable 2fa
  - login with 2fa enabled
- to test UX updates
  - login as any user
  - notice there is now a `logout` button in the top-right corner of the page / upper navbar
  - click to logout
  - login as any user
  - click the `Welcome, username` button in the top navbar
  - notice it now redirects to the user page


Note: this also redirects the Django admin login page to the Arches login page

### Issues Solved

#8022
